### PR TITLE
macos aarch64 build issue

### LINKED ISSUE #127: Failed to run custom build command for `libtor-sys v47.13.0+0.4.7.x`
### Description
When running `cargo build` with `libtor-sys 47.13.0` as dependency , the build terminate on building the `libtor-sys v47.13.0+0.4.7.x` and throwing the 
Error : 
```
process didn't exit successfully: 
  --- stdout
  OPT_LEVEL = Some("0")
  cargo:rerun-if-env-changed=CC_x86_64-apple-darwin
  CC_x86_64-apple-darwin = None
  cargo:rerun-if-env-changed=CC_x86_64_apple_darwin
  CC_x86_64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-apple-darwin
  CFLAGS_x86_64-apple-darwin = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_apple_darwin
  CFLAGS_x86_64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
```
### Environment: 
- operating system : MacOS - Darwin
- hardware Architecture : arm 64
- rustc 1.77.1 
### Reason:
The Rust project depends on libtor, a wrapper around libtor-sys written in C. Due to architecture-specific binaries, compatibility issues arise on different systems

### PR DESCRIPTION
**Pull request by 007harshmahajan**
RESOLVE #127

### CODE REVIEW

**Comment by 007harshmahajan** [comment]
and run following command for build
`cargo build --target=aarch64-apple-darwin  --features=vendored-openssl`

**Comment by Shourya742** [comment]
This solution works, but if possible, could we have a Dockerfile that supports cross-compilation? This would allow it to function not only on M1 chip Macs but also on Windows. Otherwise, we will need to continuously add libraries to mitigate build issues. 

@mojoX911, let me know your thoughts on this.

**Comment by mojoX911** [comment]
> This solution works, but if possible, could we have a Dockerfile that supports cross-compilation? This would allow it to function not only on M1 chip Macs but also on Windows. Otherwise, we will need to continuously add libraries to mitigate build issues.

Yes we should. Can you turn this into an Issue so we can track for later? 

Is this PR good to go?

**Comment by 007harshmahajan** [comment]
docker works, nice compatiblity ahead, we can close PR but ig let issue be open as working with `rust-analyser` makes local compilation, im not sure can we use it with docker instance, @Shourya742 help us on it

File: Cargo.toml [hunk 1]
```diff
@@ -19,17 +19,17 @@ dirs = "3.0.1"
...
+libtor = { version = "47.13.0", optional = true }
+mitosis = { version = "0.1.1", optional = true }
 log4rs = "1.3.0"
-
+openssl-sys = { version = "0.9.68", optional = true }
 
 #Empty default feature set, (helpful to generalise in github actions)
 [features]
 default = ['tor']
+vendored-openssl = ["openssl-sys/vendored"]
```
Comment (mojoX911) [inline Cargo.toml:30]: As discussed in the dev call, let's change this to `build-mac-os`, that's a more clear name. 

Add a comment above as a todo for adding more platform-specific build support.

We can create two different follow-up issues on of this.

- Short term: Add more platform specific fix so more devs can run it wherever they need to.
- Long term: Create a docker based solution that can run on any platform.
- Even Longer term: Ensure **Reproducible Builds** for all the apps and deamons.

For now, I am inclined to merge this as a short-term fix for #127 .

**Review by mojoX911 — CHANGES_REQUESTED** [review]

**Comment by 007harshmahajan** [comment]
UPDATE : feature tag for macos compilation 
compile with 
`cargo build --target=aarch64-apple-darwin  --features=build-macos-aarch`

**Comment by mojoX911** [comment]
> compile with
cargo build --target=aarch64-apple-darwin  --features=build-macos-aarch

We need to add this in readme too. Do they need --target if they are already compiling on native platform?

**Comment by 007harshmahajan** [comment]
will add that to readme 
no need `target` for existing compilation as rust takes this info from system calls, just need that for macos (M series)

**Review by Shourya742 — CHANGES_REQUESTED** [review]
It works fine on all macOS chipsets. Just squash the commits into a single commit.

**Review by Shourya742 — COMMENTED** [review]
@007harshmahajan Can you sign your commits?

File: Cargo.toml [hunk 2]
```diff
@@ -19,17 +19,19 @@ dirs = "3.0.1"
...
+libtor = { version = "47.13.0", optional = true }
+mitosis = { version = "0.1.1", optional = true }
 log4rs = "1.3.0"
-
+openssl-sys = { version = "0.9.68", optional = true }
 
 #Empty default feature set, (helpful to generalise in github actions)
 [features]
 default = ['tor']
+
```
Comment (Shourya742) [inline Cargo.toml:30]:
```suggestion

```

**Review by Shourya742 — CHANGES_REQUESTED** [review]
One last nit

File: Cargo.toml (hunk 2 above)
Comment (007harshmahajan) [reply Cargo.toml]: done

**Review by Shourya742 — APPROVED** [review]
ACK 70bfccf2400ce6316df5ec79c4c8218a84d1a1ce

**Review by mojoX911 — APPROVED** [review]
ACK

**Pull request merged**